### PR TITLE
Add support for more shields

### LINKED ISSUE #9: Support for all the shields from shields.io
I would love to use this package in my module called [mos](https://github.com/zkochan/mos). My module can add badges to README files automatically. I believe I can remove the logic of generating the badges from my module and use this module for that purpose.

What I would need in the shields module is:
- [ ] support for all the shields at shields.io
- [x] the ability to specify the style of the shields

I can work on it and create a pull request if your OK with these new features being part of the shields module.

### PR DESCRIPTION
**Pull request by zkochan**
closes #9 

I had to make breaking changes to the API. The CLI part works as before.

Added the possibility to specify the shield's style.

Added 6 new shields

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,19 +52,34 @@ $ npm install shields
...
-  - `obj.link` is a _String_ link to the project page of `slug` on the website
+``` js
+var shields = shielder({
+  style: 'plastic'
+})
+```
+
+### `shields(service, opts)`
+
+- `service` is a _String_ representing the service for wich the shield will be created. Currently supported values are: npm, travis, coveralls, climate, deps, devdeps, peerdeps, gemnasium.
```
Comment (KenanY) [inline README.md]: "wich" => "which"

File: bin/cmd.js [hunk 2]
```diff
@@ -48,11 +48,14 @@ readJson(p, function(error, pkg) {
   var links = [''];
 
   argv._.forEach(function(service) {
-    var shield = shields(slug, service);
-    var status = service === 'travis' ? 'Build' : 'Dependency';
-    console.log('[![' + status + ' Status][' + service + '-svg]][' + service + ']');
+    var shield = shields(service, {
+      repo: slug,
+      npmName: pkg.name,
```
Comment (KenanY) [inline bin/cmd.js]: extra comma

File: test/index.js [hunk 3]
```diff
@@ -1,18 +1,37 @@
...
-  t.ok(isFunction(shields));
+  t.ok(isFunction(shielder));
 });
 
 test('travis', function(t) {
   t.plan(1);
 
-  var s = shields('KenanY/shields', 'travis');
+  var s = shielder()('travis', {
+    repo: 'KenanY/shields',
```
Comment (KenanY) [inline test/index.js]: extra comma

File: test/index.js [hunk 4]
```diff
@@ -1,18 +1,37 @@
...
+    image: 'https://img.shields.io/travis/KenanY/shields.svg?style=flat',
+    link: 'https://travis-ci.org/KenanY/shields'
+  });
+});
+
+test('flat square styled travis', function(t) {
+  t.plan(1);
+
+  var shields = shielder({
+    style: 'flat-square',
```
Comment (KenanY) [inline test/index.js]: extra comma

File: test/index.js [hunk 5]
```diff
@@ -1,18 +1,37 @@
...
+  });
+  var s = shields('travis', {
+    repo: 'KenanY/shields',
```
Comment (KenanY) [inline test/index.js]: extra comma

File: lib/shields-list.js [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    text: 'peerDependency status',
+    path: 'david/peer/{repo}',
+    link: 'https://david-dm.org/{repo}#info=peerDependencies'
+  },
+  gemnasium: {
+    text: 'Gemnasium',
+    path: 'gemnasium/{repo}',
+    link: 'https://gemnasium.com/{repo}'
+  },
+}
```
Comment (KenanY) [inline lib/shields-list.js]: semicolon

File: lib/shields-list.js [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  peerdeps: {
+    text: 'peerDependency status',
+    path: 'david/peer/{repo}',
+    link: 'https://david-dm.org/{repo}#info=peerDependencies'
+  },
+  gemnasium: {
+    text: 'Gemnasium',
+    path: 'gemnasium/{repo}',
+    link: 'https://gemnasium.com/{repo}'
+  },
```
Comment (KenanY) [inline lib/shields-list.js]: extra comma

File: test/index.js [hunk 8]
```diff
@@ -20,9 +39,12 @@ test('travis', function(t) {
 test('gemnasium', function(t) {
   t.plan(1);
 
-  var s = shields('KenanY/shields', 'gemnasium');
+  var s = shielder()('gemnasium', {
+    repo: 'KenanY/shields',
```
Comment (KenanY) [inline test/index.js]: extra comma

**Comment by zkochan** [comment]
@KenanY thanks!

Consider adding an eslinrc file to the project, so that people know what standards they should follow

**Comment by KenanY** [comment]
Good idea :)

File: lib/shields-list.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
+'use strict'
+module.exports = {
+  npm: {
+    text: 'NPM version',
+    path: 'npm/v/{npmName}',
+    link: 'https://npmjs.org/package/{npmName}'
```
Comment (zkochan) [inline lib/shields-list.js]: oops, this one should be www.npmjs.com

**Comment by KenanY** [comment]
Landed as https://github.com/KenanY/shields/commit/dc8eb1dde816ac088bc8b4ec20521cbac35c3356 but looks like I just missed your latest commit. Maybe I'll amend the commit.

**Comment by zkochan** [comment]
Yeah, sorry for that :sweat_smile:

**Comment by KenanY** [comment]
Here it is https://github.com/KenanY/shields/commit/b1ca359f949f4b858811b711fb87e2df78400378 :)

**Pull request closed without merging**